### ISSUE TITLE
Scoring for metrics and losses, 

### ISSUE THREAD

**Issue by EngrStudent** [keras-team/keras#15782]
Disclaimer: I'm an engineer by training, not a statistician, so I can get this wrong.  I don't know that the general thrust has technical gaps, but I could be off in some of the details.  If I am, please go talk to your friendly neighborhood stats professor and get their thoughts on the idea. 

**Background:**    
I hang out (and interact, and have for years, and learn tons) at [CrossValidated](https://stats.stackexchange.com/), a stack-exchange site.

One of the very substantial families of threads there is why accuracy is not ideal in many places, and many of the folks engaged in the discussions are fantastic PhD's, in academia and industry, teaching or working for decades, so they are a very important source of technical wisdom.

Here are some of the threads there:

- https://stats.stackexchange.com/questions/359909/is-accuracy-an-improper-scoring-rule-in-a-binary-classification-setting
- https://stats.stackexchange.com/questions/357466/are-unbalanced-datasets-problematic-and-how-does-oversampling-purport-to-he
- https://stats.stackexchange.com/questions/312780/why-is-accuracy-not-the-best-measure-for-assessing-classification-models
- https://stats.stackexchange.com/questions/359909/is-accuracy-an-improper-scoring-rule-in-a-binary-classification-setting/359936#359936
- **https://stats.stackexchange.com/questions/368949/example-when-using-accuracy-as-an-outcome-measure-will-lead-to-a-wrong-conclusio**
  

They like these things called ["strictly proper score functions" or "strictly proper scoring rules"](https://www.wikiwand.com/en/Scoring_rule#/StrictlyProperScoringRules).  

Here are references on strictly proper scoring rules:

- https://apps.dtic.mil/sti/pdfs/ADA459827.pdf
- (of course) https://en.wikipedia.org/wiki/Scoring_rule#Proper_scoring_rules
- (other folks in github have engaged them) https://github.com/mlr-org/mlr/issues/880
- https://sites.stat.washington.edu/raftery/Research/PDF/Gneiting2007jasa.pdf
- https://www.tensorflow.org/probability/api_docs/python/tfp/stats/brier_score
- https://xianblog.wordpress.com/2017/11/21/the-hyvarinen-score-is-back/
- https://faculty.missouri.edu/~merklee/pub/MerkleSteyvers2013.pdf

When I got to keras loss and metrics pages I don't see any of those scoring rules, and I think its a miss.

Current losses from [documentation](https://keras.io/api/losses/):

- binary/categorical cross-entropy
- KL divergence
- Poisson class

Current metrics from [documentation](https://keras.io/api/metrics/):

- Accuracy
- Binary/Categorical/TopK accuracy 
- Binary/categorical crossentropy
- AUC/and the TF PN measures

**Recommendation/Suggestion:**     
I think you should add the following "strictly proper scoring rules" to Keras because it can make it easier for new users (and their pointy-haired bosses) to use technically exemplary approaches in some of their problem solving.

Some rules to consider:

- [Brier](https://en.wikipedia.org/wiki/Brier_score)/quadratic scoring rule 
- [Hyvarinen scoring rule](https://xianblog.wordpress.com/2017/11/21/the-hyvarinen-score-is-back/)
- Spherical Scoring Rule
- Logarithmic scoring rule (log-probability)

**Comment by qlzh727** [comment]
Triage notes: we might want to add a keras.io example for how to choose a proper loss/metric.

**Comment by EngrStudent** [comment]
This answer gives references for examples of using strictly proper scoring rules:    
https://stats.stackexchange.com/questions/279053/how-to-see-how-good-my-probability-estimations-are/279056#279056

Here are some of the references it gives:

- https://www.tandfonline.com/doi/abs/10.1198/016214506000001437
- https://www.tandfonline.com/doi/abs/10.1198/jbes.2010.08110
- https://www.sciencedirect.com/science/article/abs/pii/S0304407611000807

**Comment by EngrStudent** [comment]
@old-school-kid  you are correct, sir.

**Comment by old-school-kid** [comment]
@EngrStudent in that case I think it would be a good fit under tensorflow probability. They already have `brier_decomposition` and a family of logarithmic scoring rules.

**Comment by EngrStudent** [comment]
Going fast is critical, but if you go fast in the wrong way in your car you can crash into a wall, and that is a bad thing.  Keras should be both fast AND useful, and in this case by useful I mean correct instead of misleading, which the wrong measures of goodness can easily become.
